packages/markdown-it-plugins/src/index.ts

```diff
@@ -0,0 +1,43 @@
+import type MarkdownIt from "markdown-it";
+import { render } from "./render.ts";
+import { ENV_KEY } from "./util.ts";
+import { hookTemplateLiterals } from "./template-literal.ts";
+import { hookFence } from "./fence.ts";
+
+function hookRender(md: MarkdownIt) {
+    const originalRender = md.render;
+    md.render = (src: string, _env?: any): string => {
+        const env = { ..._env };
+        const retVal = originalRender.call(md, src, env);
+        render(env[ENV_KEY] ?? []);
+        return retVal;
+    };
+}
+
+function hookVitepressRender(md: MarkdownIt) {
+
+    const originalRender = md.render;
+    md.render = (src: string, env?: any): string => {
+        const myEnv = env ?? {};
+        let retVal = originalRender.call(md, src, myEnv);
```

Comment:
why's this different from how hookRender does the env stuff above?

Reply:
Up to that point both are the same (and I could re-structure to make cleaner), its how the Observable render gets triggered that is different depending on the host environment.